File: algorithms/sort/bubble/bubble.java
import java.util.Arrays;
import java.util.Scanner;


class Bubble {
    public static void main(String [] args) {

        Scanner in = new Scanner(System.in);

        System.out.print("Length: ");

        int length = in.nextInt();

        int[] array = new int[length];

        for (int i = 0; i < length; i++) {
            System.out.printf("Item %d: ", i);
            array[i] = in.nextInt();
        }

        System.out.println("Source array " + Arrays.toString(array));

        boolean loop = true;

        while (loop) {
            loop = false;
            for (int i = 0; i < length - 1; i++) {
                if (array[i] > array[i + 1]) {
                    loop = true;
                    int tmp = array[i];
                    array[i] = array[i + 1];
                    array[i + 1] = tmp;
                }
            }
        }

        System.out.println("Sorted array " + Arrays.toString(array));

        in.close();
    }
}

File: algorithms/sort/bubble/bubble.c
#include <stdio.h>
#include <stdbool.h>


void print_array(char * pre, int array[], int length)
{

    printf("%s[ ", pre);

    for (int i = 0; i < length; i++) {
        if (i != length - 1)
        {
            printf("%d,", array[i]);
        }
        else
        {
            printf("%d ]\n", array[i]);
        }
    }
}

int main()
{

    int length;

    printf("Length: ");
    scanf("%d", &length);

    int array[length];

    for (int i = 0; i < length; i++)
    {
        printf("Item %d: ", i);
        scanf("%d", &array[i]);
    }

    print_array("Source array: ", array, length);

    bool loop = true;

    while (loop)
    {
        loop = false;
        for (int i = 0; i < length - 1; i++)
        {
            if (array[i] > array[i + 1])
            {
                loop = true;
                int tmp = array[i];
                array[i] = array[i + 1];
                array[i + 1] = tmp;
            }
        }
    }
    
    print_array("Sorted array: ", array, length);

    return 0;
}